app/src/main/java/com/example/music_app/ui/models/classes/AppLoginWebViewClient.kt

```diff
@@ -0,0 +1,15 @@
+package com.example.music_app.ui.models.classes
+
+import android.net.Uri
+import android.webkit.WebView
+import android.webkit.WebViewClient
+
+class AppLoginWebViewClient : WebViewClient() {
+    override fun onPageFinished(view: WebView?, url: String?) {
+        super.onPageFinished(view, url)
+        val uri = Uri.parse(url)
+        val state = uri.getQueryParameter("state")
+        val error = uri.getQueryParameter("error")
```

Comment:
Inside of the separate method query parameter extracting could be simplified with using `with`
```suggestion
        with(uri) {
        ....
        val error = getQueryParameter("error")
        ....
        }
```